### PR TITLE
18 support public channels

### PR DESCRIPTION
**Pull request by dominic22**
# Description of change
- support public channels in ui
- add route to see history of a channel
- add route to set options to receive the history of a channel

## Links to any relevant issues
https://github.com/iotaledger/is-ui-components/issues/18

## Type of change

-   Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

> You need to run you own local version of the Integration Services. Make sure you have used the latest `develop` branch.

#### public channel history
click on history tab in navigation bar and add a channel address in it as for instance:
channeladdress: `cb7f036187a52053a9520bfe45d6521714bd7bd3308213e79fa7ce92c68cc42d0000000000000000:1ae5fa591c61c36eb2b0ae40`
- use `public` for channel type
see: http://localhost:3044/history/cb7f036187a52053a9520bfe45d6521714bd7bd3308213e79fa7ce92c68cc42d0000000000000000:1ae5fa591c61c36eb2b0ae40?type=public

#### private channel history

channelAddress: `4abdccf654513935408a5d46aaec6fc4e23c658db50676b1c30a88122273a3a70000000000000000:8860ed8d5a948632aef486ec`

presharedKey: `4212ae4d1e375be0c8f97864c0419058`
- use `private` for channel type
see: http://localhost:3044/history/4abdccf654513935408a5d46aaec6fc4e23c658db50676b1c30a88122273a3a70000000000000000:8860ed8d5a948632aef486ec?type=private&preshared-key=4212ae4d1e375be0c8f97864c0419058

## Change checklist

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have checked that new and existing unit tests pass locally with my changes
-   [ ] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by JuriBog — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "name": "@iota/is-ui-components",
     "type": "module",
-    "version": "0.0.44",
+    "version": "0.0.45",
     "description": "Svelte UI components for IOTA Integration Services",
     "scripts": {
-        "dev": "svelte-kit dev --port ${PORT:-3000}",
+        "dev": "svelte-kit dev --port ${PORT:-3044}",
```
Comment (mastrogiovanni) [inline package.json:7]: Set default port to 3000 (as convention)

File: src/lib/components/streams/channel-details.svelte [hunk 2]
```diff
@@ -21,6 +22,12 @@
 
     async function manageChannelData(): Promise<void> {
         if (subscriptionStatus === SubscriptionState.Authorized) {
+            if (channel.type === ChannelType.public) {
```
Comment (mastrogiovanni) [inline src/lib/components/streams/channel-details.svelte:25]: Do public channel can be readed also when subscription status if not SubscriptionState.Authorized?

**Review by JuriBog — APPROVED** [review]

**Review by mastrogiovanni — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (dominic22) [reply package.json:7]: no it will block since port already is used

File: src/lib/components/streams/channel-details.svelte (hunk 2 above)
Comment (dominic22) [reply src/lib/components/streams/channel-details.svelte:25]: only on the /history route for the detail route it is only possible if authorized to have a consistent behavior

**Review by JuriBog — APPROVED** [review]

**Pull request merged**